### PR TITLE
Added the possibility to define more protocol IDs for p2p networks

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- added the possibility to define more protocol IDs for p2p networks
  
related to https://github.com/multiversx/mx-chain-communication-go/pull/48

## Proposed changes
- 
- 
- 

## Testing procedure
- start a system testnet. Make sure that when we upgrade the nodes (epoch 1) the seednode is compiled from this branch and the p2p.toml file for the seeder is fetched from this branch. Make sure that after the upgrade process for all nodes their p2p.toml is fetched from this branch. The test should process blocks normally
- start a half-half test with this branch and `rc/v1.7.next` branch. The seeder must be compiled from this branch & its p2p.toml should be used from this branch. All nodes upgraded to this branch variant should use their p2p.toml fetched from this branch.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/node/config/p2p.toml [hunk 1]
```diff
@@ -48,9 +48,12 @@
     # RefreshIntervalInSec represents the time in seconds between querying for new peers
     RefreshIntervalInSec = 10
 
-    # ProtocolID represents the protocol that this node will advertize to other peers
-    # To connect to other nodes, those nodes should have the same ProtocolID string
-    ProtocolID = "/erd/kad/1.0.0"
+    # ProtocolIDs represents the protocols that this node will advertize to other peers
+    # To connect to other nodes, those nodes should have at least on common protocol string
```
Comment (sstanculeanu) [inline cmd/node/config/p2p.toml:52]:
```suggestion
    # To connect to other nodes, those nodes should have at least one common protocol string
```

File: cmd/node/config/fullArchiveP2P.toml [hunk 2]
```diff
@@ -48,9 +48,12 @@
     # RefreshIntervalInSec represents the time in seconds between querying for new peers
     RefreshIntervalInSec = 10
 
-    # ProtocolID represents the protocol that this node will advertize to other peers
-    # To connect to other nodes, those nodes should have the same ProtocolID string
-    ProtocolID = "/erd/kad/1.0.0"
+    # ProtocolIDs represents the protocols that this node will advertize to other peers
+    # To connect to other nodes, those nodes should have at least on common protocol string
```
Comment (sstanculeanu) [inline cmd/node/config/fullArchiveP2P.toml:52]:
```suggestion
    # To connect to other nodes, those nodes should have at least one common protocol string
```

File: cmd/seednode/config/p2p.toml [hunk 3]
```diff
@@ -47,9 +47,12 @@
     #RefreshIntervalInSec represents the time in seconds between querying for new peers
     RefreshIntervalInSec = 10
 
-    #ProtocolID represents the protocol that this node will advertize to other peers
-    #To connect to other nodes, those nodes should have the same ProtocolID string
-    ProtocolID = "/erd/kad/1.0.0"
+    # ProtocolIDs represents the protocols that this node will advertize to other peers
+    # To connect to other nodes, those nodes should have at least on common protocol string
```
Comment (sstanculeanu) [inline cmd/seednode/config/p2p.toml:51]:
```suggestion
    # To connect to other nodes, those nodes should have at least one common protocol string
```

File: config/tomlConfig_test.go [hunk 4]
```diff
@@ -498,7 +499,13 @@ func TestP2pConfig(t *testing.T) {
     Enabled = false
     Type = ""
     RefreshIntervalInSec = 0
-    ProtocolID = "` + protocolID + `"
+
+    # ProtocolIDs represents the protocols that this node will advertize to other peers
+    # To connect to other nodes, those nodes should have at least on common protocol string
```
Comment (sstanculeanu) [inline config/tomlConfig_test.go:504]:
```suggestion
    # To connect to other nodes, those nodes should have at least one common protocol string
```

File: cmd/node/config/p2p.toml (hunk 1 above)
Comment (iulianpascalau) [reply cmd/node/config/p2p.toml:52]: fixed

File: cmd/node/config/fullArchiveP2P.toml (hunk 2 above)
Comment (iulianpascalau) [reply cmd/node/config/fullArchiveP2P.toml:52]: fixed

File: cmd/seednode/config/p2p.toml (hunk 3 above)
Comment (iulianpascalau) [reply cmd/seednode/config/p2p.toml:51]: fixed

File: config/tomlConfig_test.go (hunk 4 above)
Comment (iulianpascalau) [reply config/tomlConfig_test.go:504]: fixed

**Review by AdoAdoAdo — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
-- Allin ✅ 
-- Half-Half ✅ 
-- p2p.toml from branch ✅ 


Half - Half test: rc/v1.7.next1 and more-protocol-ids-9c29760c9a

--- Specific errors ---

block hash does not match 2814
wrong nonce in block 1286
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  1
Nr. of all WARNS:  204
Nr. of new ERRORS:  1
Nr. of new WARNS:  0
Nr. of PANICS:  0

/------/

--- ERRORS ---

upcloud-fra-validator-5 :
Error: bootstrapDataComponentsFactory create() failed: bootstrap process has failed: time is out  1

/------/

--- WARNINGS ---

/------/

**Pull request merged**